fe/fe-core/src/main/java/org/apache/doris/load/loadv2/IngestionLoadJob.java

```diff
@@ -0,0 +1,1138 @@
+        descTable.computeStatAndMemLayout();
+        return descTable.toThrift();
+    }
+
+    private PushTask buildPushTask(long backendId, OlapTable olapTable, long taskSignature, long partitionId,
+                                   long indexId, long tabletId, long replicaId, int schemaHash, int schemaVersion,
+                                   long bucket)
+            throws AnalysisException {
+
+        DescriptorTable descTable = new DescriptorTable();
+        TupleDescriptor destTupleDesc = descTable.createTupleDescriptor();
+
+        List<TColumn> columnsDesc = new ArrayList<>();
+        List<Column> columns = new ArrayList<>();
+        for (Column column : olapTable.getSchemaByIndexId(indexId)) {
+            Column col = new Column(column);
+            col.setName(column.getName().toLowerCase(Locale.ROOT));
+            columns.add(col);
+            columnsDesc.add(col.toThrift());
+            // use index schema to fill the descriptor table
+            SlotDescriptor destSlotDesc = descTable.addSlotDescriptor(destTupleDesc);
+            destSlotDesc.setIsMaterialized(true);
+            destSlotDesc.setColumn(col);
+            destSlotDesc.setIsNullable(col.isAllowNull());
+        }
+
+        // deep copy TBrokerScanRange because filePath and fileSize will be updated
+        // in different tablet push task
+        TBrokerScanRange tBrokerScanRange =
+                getTBrokerScanRange(descTable, destTupleDesc, columns, hadoopProperties);
+        // update filePath fileSize
+        TBrokerRangeDesc tBrokerRangeDesc = tBrokerScanRange.getRanges().get(0);
+        tBrokerRangeDesc.setFileType(TFileType.FILE_HDFS);
+        tBrokerRangeDesc.setPath("");
+        tBrokerRangeDesc.setFileSize(-1);
+        String tabletMetaStr = String.format("%d.%d.%d.%d.%d", olapTable.getId(), partitionId,
+                indexId, bucket, schemaHash);
+        if (tabletMetaToFileInfo.containsKey(tabletMetaStr)) {
+            Pair<String, Long> fileInfo = tabletMetaToFileInfo.get(tabletMetaStr);
+            tBrokerRangeDesc.setPath(fileInfo.first);
+            tBrokerRangeDesc.setFileSize(fileInfo.second);
+        }
+
+        TDescriptorTable tDescriptorTable = getTDescriptorTable(descTable);
+
+        return new PushTask(backendId, dbId, olapTable.getId(),
+                partitionId, indexId, tabletId, replicaId, schemaHash, 0, id,
+                TPushType.LOAD_V2, TPriority.NORMAL, transactionId, taskSignature,
+                tBrokerScanRange, tDescriptorTable, columnsDesc,
+                olapTable.getStorageVaultId(), schemaVersion);
+    }
+
+    public void updateLoadingStatus() throws UserException {
+        if (!checkState(JobState.LOADING)) {
+            return;
+        }
+
+        if (etlStatus.getState() == TEtlState.CANCELLED) {
+            throw new LoadException(etlStatus.getFailMsg());
+        }
+
+        // submit push tasks
+        Set<Long> totalTablets = submitPushTasks();
```

Comment:
How to make sure that push tasks will NOT be submitted multi-times?
Because the `updateLoadingStatus()` may be called many times.

Reply:
`tabletToSentReplicaPushTask` records the tablets that have been sent and the push tasks of the replicas. Before building a push task, this map will be used to check whether it has been sent.